### PR TITLE
Adding CWE-400 README.md and updated main README.md

### PR DESCRIPTION
**Pull request by myteron**

### CODE REVIEW

File: CWE-664/CWE-400/README.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+## Related Guidelines
+
+|||
+|:---|:---|
+|[MITRE CWE](http://cwe.mitre.org/)|Pillar [CWE-664: Improper Control of a Resource Through its Lifetime (4.13) (mitre.org)](https://cwe.mitre.org/data/definitions/664.html)|
+|[MITRE CWE](http://cwe.mitre.org/)|Class [CWE-400: Uncontrolled Resource Consumption (4.12)](https://cwe.mitre.org/data/definitions/400.html)|
+|[SEI CERT Coding Standard for Java](https://wiki.sei.cmu.edu/confluence/display/java/SEI+CERT+Oracle+Coding+Standard+for+Java)|[TPS02-J. Ensure that tasks submitted to a thread pool are interruptible](https://wiki.sei.cmu.edu/confluence/display/java/TPS02-J.+Ensure+that+tasks+submitted+to+a+thread+pool+are+interruptible)|
+
+## Biblography
```
Comment (s19110) [inline CWE-664/CWE-400/README.md:102]: Typo in Bibliography (missing i)

File: CWE-664/CWE-400/README.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+|||
+|:---|:---|
+|[[Python 3.10.4 docs Future.cancel]](https://docs.python.org/)|concurrent.futures — Launching parallel tasks — Python 3.10.4 documentation. Available from: <https://docs.python.org/3/library/concurrent.futures.html> \[Last Accessed May 2024]|
```
Comment (s19110) [inline CWE-664/CWE-400/README.md:106]: Wrong URL in hyperlink (should be https://docs.python.org/3/library/concurrent.futures.html instead.)

File: CWE-664/CWE-400/README.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+|[[Python 3.10.4 docs on threading.Event]](https://docs.python.org/)|threading — Thread-based parallelism - Event Objects. Available from: <https://docs.python.org/3/library/threading.html#event-objects> \[Last Accessed May 2024]|
```
Comment (s19110) [inline CWE-664/CWE-400/README.md:107]: Wrong URL in hyperlink (should be https://docs.python.org/3/library/threading.html#event-objects instead.)

File: CWE-664/CWE-400/README.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+with ThreadPoolExecutor() as executor:
+    task = run_thread(executor, "A")
+    interrupt(task)
+
+```
+
+## Compliant Solution
+
+Tasks submitted to the ThreadPoolExecutor can be interrupted by setting a thread-safe flag, such as `threading.Event`  [Python 3.10 docs on threading.Event]. An Event object should be passed as an argument to the submitted task. From within the task function, we need to manually check the flag status by calling `event.is_set()` and handling the interruption. In order to set the Event flag, we can call `event.set()` on the event object.
```
Comment (s19110) [inline CWE-664/CWE-400/README.md:48]: [Python 3.10 docs on threading.Event] doesn't link to the docs.

File: CWE-664/CWE-400/README.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
+# CWE-400: Uncontrolled Resource Consumption
+
+Canceling the task in a thread pool only prevents it from being executed if it has not started yet. For the task to be interruptible, it must handle the `threading.Event` flag.
+
+## Non-Compliant Code Example
+
+Tasks can be submitted to the ThreadPoolExecutor by calling `submit()`. Submitted tasks can be canceled by calling `cancel()` on the Future object returned by `submit()`. Calling this method will return True and stop the task from being executed if it has not started yet. However, if its execution has already started, calling `cancel()` will instead return  False and will not stop the task [Python 3.10 docs on Future.cancel].
```
Comment (s19110) [inline CWE-664/CWE-400/README.md:7]: [Python 3.10 docs on Future.cancel] doesn't link to the docs

**Review by tommcd — COMMENTED** [review]
I think it will be easy to switch to use footnotes for references as used by openssf.
Maybe we should try that?

https://github.com/ossf/wg-best-practices-os-developers/blob/main/docs/Compiler-Hardening-Guides/Compiler-Options-Hardening-Guide-for-C-and-C%2B%2B.md

**Comment by s19110** [comment]
I have resolved the comments and merge conflicts. I am squashing the changes and merging them to the master branch.

**Pull request merged**